zio-http/src/main/scala/zio/http/ServerConfig.scala

```diff
@@ -0,0 +1,182 @@
+    val (eventLoopGroupLayer, serverChannelFactoryLayer) = config.channelType match {
+      case ChannelType.NIO => (EventLoopGroup.nio(config.nThreads), ServerChannelFactory.nio)
+      case ChannelType.EPOLL => (EventLoopGroup.epoll(config.nThreads), ServerChannelFactory.epoll)
+      case ChannelType.KQUEUE => (EventLoopGroup.kQueue(config.nThreads), ServerChannelFactory.kQueue)
+      case ChannelType.URING => (EventLoopGroup.uring(config.nThreads), ServerChannelFactory.uring)
+      case ChannelType.AUTO => (EventLoopGroup.auto(config.nThreads), ServerChannelFactory.auto)
+    }
+    val configLayer: ULayer[Config] = ZLayer.succeed(config)
+    configLayer ++ eventLoopGroupLayer ++ serverChannelFactoryLayer
+  }
+
+  val testConfig: ZLayer[Any, Nothing, Config with EventLoopGroup with ServerChannelFactory] = {
+    val configLayer: ULayer[Config] = ZLayer.succeed(Config().withPort(0).withLeakDetection(LeakDetectionLevel.PARANOID))
+    configLayer ++ EventLoopGroup.nio(0) ++ ServerChannelFactory.nio
+  }
+
+  sealed trait ChannelType
+
+  object ChannelType {
+    case object NIO extends ChannelType
+
+    case object EPOLL extends ChannelType
+
+    case object KQUEUE extends ChannelType
+
+    case object URING extends ChannelType
+
+    case object AUTO extends ChannelType
+  }
+
+  sealed trait LeakDetectionLevel {
+    self =>
+    def jResourceLeakDetectionLevel: ResourceLeakDetector.Level = self match {
+      case LeakDetectionLevel.DISABLED => ResourceLeakDetector.Level.DISABLED
+      case LeakDetectionLevel.SIMPLE => ResourceLeakDetector.Level.SIMPLE
+      case LeakDetectionLevel.ADVANCED => ResourceLeakDetector.Level.ADVANCED
+      case LeakDetectionLevel.PARANOID => ResourceLeakDetector.Level.PARANOID
+    }
+  }
+
+  object LeakDetectionLevel {
+    case object DISABLED extends LeakDetectionLevel
+
+    case object SIMPLE extends LeakDetectionLevel
+
+    case object ADVANCED extends LeakDetectionLevel
+
+    case object PARANOID extends LeakDetectionLevel
+  }
+
+  final case class Config(
+                           leakDetectionLevel: LeakDetectionLevel = LeakDetectionLevel.SIMPLE,
+                           error: Option[Throwable => ZIO[Any, Nothing, Unit]] = None,
+                           sslOption: ServerSSLOptions = null,
+                           address: InetSocketAddress = new InetSocketAddress(8080),
+                           acceptContinue: Boolean = false,
+                           keepAlive: Boolean = true,
+                           consolidateFlush: Boolean = false,
+                           flowControl: Boolean = true,
+                           channelInitializer: ChannelPipeline => Unit = null,
+                           requestDecompression: (Boolean, Boolean) = (false, false),
+                           objectAggregator: Int = 1024 * 100,
+                           serverBootstrapInitializer: ServerBootstrap => Unit = null,
```

Comment:
Oops, another function. I think we need to move this out and into wire-up. If the server bootstrap needs to be customized, we can put it into a layer, and then expose a more customizable live `Server` layer which requires you to feed it a (customized) server bootstrap.

Reply:
Removed for now and added a TODO in the place where it is used. As far as I'm aware is not used.